source/main.py

```diff
@@ -1,177 +1,19 @@
-
-    #Triangle
-    if len(approx) == 3:
-        colorName = setNameShape("Triangle", int(x), int(y), img)
-
-        #Appending instances to listOfShapes
-        #Triangle = Guitar sound = number 30
-        #listOfShapes.append(Image("Triangle", int(shapeSize), colorName, int(width), int(height)))
-        listOfShapes.append(Image(30, int(shapeSize), 120, 1, 50))
-
-    #Square or rectangle
-    elif len(approx) == 4 : 
-        x2, y2 , w, h = cv2.boundingRect(approx)
-        aspectRatio = float(w)/h
-        if aspectRatio >= 0.95 and aspectRatio < 1.05:
-            colorName = setNameShape("Square",int(x), int(y), img)
-            #Square = Drum = number 119
-            listOfShapes.append(Image(119, int(shapeSize), 60, 2, 200))
-        else:
-            colorName = setNameShape("Rectangle",int(x), int(y), img)
-            #Rectangle = half circle = flute = number 74
-            listOfShapes.append(Image(74, int(shapeSize), 30, 1, 240))
-
-    #Pentgaon
-    elif len(approx) == 5 :
-        colorName = setNameShape("Pentagon",int(x), int(y), img)
-        #Pentagon = Heartshape = piano = 2
-        listOfShapes.append(Image(3, int(shapeSize), 240, 2, 100))
-
-    #Star
-    elif len(approx) == 10 :
-        colorName = setNameShape("Star",int(x), int(y), img)
-        #Star = Cello = 43
-        listOfShapes.append(Image(43, int(shapeSize), 90, 1, 150))
-
-    #Circle
-    else:
-        colorName = setNameShape("Circle",int(x), int(y),img)
-        #Circle = lead 1 = 81
-        listOfShapes.append(Image(81, int(shapeSize), 120, 1, 200))
-
-# Accessing object value using a for loop
-for shape in listOfShapes:
-    print("instrument:", shape.instrument, "volume:", shape.volume, "bpm:", shape.bpm, "pitch:", shape.pitch, "duration:", shape.duration, sep='\t')
-    
-while(1):
-    cv2.imshow("image",img)
-
-    #Break the loop when user hits 'esc' key    
-    if cv2.waitKey(20) & 0xFF ==27:
-        break
-    
-cv2.destroyAllWindows()
-
-def MakeSong(list):
-    #pitch, bpm, duration, volume, instrument, amount
-    amount_of_instruments = len(list)
-    for shape in list:
-
-        # create MIDIFile object
-        midi = MIDIFile(amount_of_instruments, removeDuplicates=False)
-
-        # add tracks
-        track1 = 0
-        time = 0
-        channel = 0 
-        instruments = 0
-
-        while instruments < amount_of_instruments:
-        # create ass many tracks as instruments        
-            midi.addTrackName(instruments, time, f"Track{instruments}")
-            midi.addTempo(instruments, time, shape.bpm)
-            midi.addProgramChange(instruments, 0, time, shape.instrument)
-
-            midi.addNote(track1, channel, shape.pitch, time, shape.duration, shape.volume)
-            time = +2
-            instruments +=1
-
-        with open("output.mid", "wb") as output_file:
-            midi.writeFile(output_file)
+        #Break the loop when user hits 'esc' key    
```

Comment:
Oh, THat is why it did not create sound.  
I did not know i had to press the `esc` key to continue...  
maybe a `print('Press esc to continue...')` could help here.

Reply:
Good point. will add it to the code.